### ISSUE TITLE
sw_engine: vector-effect non-scaling-stroke not supported

### ISSUE THREAD

**Issue by lostminds** [thorvg/thorvg#1925]
SVG by default scales the stroke width of transformed elements. This is sometimes not desired, especially with non-uniform scaling or skewing. For this reason a special [vector-effect: non-scaling-stroke](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/vector-effect#non-scaling-stroke) property exists to make stroke width not scale with transform on elements. However, it doesn't seem like it supported currently by ThorVG:

![thorvg0 12 2 stroke scaling test no scale jpg](https://github.com/thorvg/thorvg/assets/17763524/c4bbc4fe-0710-40c8-8936-b46614c977e5)
(Left is Firefox, right ThorVG viewer 0.12.2)

The file used:
![StrokeScaleTest non-scaling-stroke](https://github.com/thorvg/thorvg/assets/17763524/dcd30235-2f79-4c62-977c-f7dd5a9f06d3)

As you can see there are also some issues with the rendering of the scaled stroke due to https://github.com/thorvg/thorvg/issues/1915

**Comment by capnm** [comment]
Yes, it's not supported and the path also has the matrix scaling issue.
```xml
<svg xmlns="http://www.w3.org/2000/svg"
  width="500px" height="400px" viewBox="0 0 500 400">
<g id="Artboard">
<path id="Circle" 
  d="M 0 -46.5 
    C 25.6812408671317 -46.5, 46.5 -25.6812408671317,46.5 0
    C 46.5 25.6812408671317, 25.6812408671317 46.5,0 46.5
    C -25.6812408671317 46.5, -46.5 25.6812408671317,-46.5 0
    C -46.5 -25.6812408671317, -25.6812408671317 -46.5, 0 -46.5
    Z" 
  fill="rgb(190,190,190)" 
  stroke="rgb(0,0,0)" 
  stroke-width="8" 
  transform="matrix(4,0,0,0.5,250,200)" 
  vector-effect="non-scaling-stroke"
/>
</g>
</svg>
```
```
[L] SVG (../thorvg-git/src/loaders/svg/tvgXmlParser.cpp 357):
  Unsupported attributes used [Elements type: Path] [Id : Circle]
     [Attribute: vector-effect] [Value: non-scaling-stroke]
```
GNOME image viewer, LibreOffice: 
![LibreOffice](https://github.com/thorvg/thorvg/assets/4047289/39acf01c-3fdb-444e-978b-e21d51e363e4)

**Comment by hermet** [comment]
sw / gl / wgpu:

![Image](https://github.com/user-attachments/assets/83f627af-27fa-401d-b71b-866f6fede5f7)

**Comment by hermet** [comment]
@lostminds @capnm Hello, the fix will be in v1.0.6, Thanks!
